ci: intermittent selenium test

As seen [in this github workflow run failure](https://github.com/jupyterhub/jupyterhub/actions/runs/3515164623/jobs/5890079884#step:12:115), a test is failing. But it does so intermittently.

```python-traceback
async def test_open_url_login(
        # ...
        else:
            if next_url.endswith(f"/user/{user}/") == False:
                await webdriver_wait(
                    browser, EC.url_to_be(ujoin(public_url(app), f"/user/{user}/"))
                )
>           assert next_url.endswith(f"/user/{user}/")
E           AssertionError: assert False
E            +  where False = <built-in method endswith of str object at 0x7f5445c9c530>('/user/test_user/')
E            +    where <built-in method endswith of str object at 0x7f5445c9c530> = 'http://127.0.0.1:33[92](https://github.com/jupyterhub/jupyterhub/actions/runs/3515164623/jobs/5890079884#step:12:93)7/@/space%20word/hub/spawn-pending/test_user'.endswith
```

I'm not sure what the fix is for this test, but perhaps its relates to a configuration option to automatically spawn a server after logging in or not? Where we end up getting redirected to the spawn pending page.

--- a/jupyterhub/tests/selenium/test_browser.py
+++ b/jupyterhub/tests/selenium/test_browser.py
@@ -88,9 +88,8 @@ def clear(browser, by_locator):
 # LOGIN PAGE
 async def test_elements_of_login_page(app, browser):
     await open_url(app, browser)
-    logo = is_displayed(browser, LoginPageLocators.LOGO)
+    assert is_displayed(browser, LoginPageLocators.LOGO)
     logo_text = browser.find_element(*LoginPageLocators.LOGO).get_attribute("innerHTML")
-    assert logo == True
 
 
 async def login(browser, user, pass_w):
@@ -186,10 +185,11 @@ async def test_open_url_login(
         assert next_url.endswith("spawn?param=value")
         assert f"user/{user}/" not in next_url
     else:
-        if next_url.endswith(f"/user/{user}/") == False:
+        if not next_url.endswith(f"/user/{user}/"):
             await webdriver_wait(
                 browser, EC.url_to_be(ujoin(public_url(app), f"/user/{user}/"))
             )
+            next_url = browser.current_url
         assert next_url.endswith(f"/user/{user}/")
 
 

